src/pkg/caendr/caendr/models/datastore/cart.py

```diff
@@ -35,29 +32,27 @@ def get_props_set(cls):
       *super().get_props_set(),
       'user',
       'items',
-      'is_deleted',
       'version'
     }
   
   
 
   @classmethod
   def lookup_by_user(cls, email):
-
-    # Try finding and returning unique card for this user
+    
+    # Try finding cart(s) for user
     try:
-      return Cart.query_ds_not_deleted('user', email, required=True)
+      # Run the query and try to find not deleted cart(s) for the user
+      carts = cls.query_ds_not_deleted('user', email, required=True)
 
+      # If one cart was found, return it
+      # Or if multiple carts were found, sort by date modified on and return latest
+      return cls.sort_by_modified_date(carts, reverse=True)[0]
 
-    # User doesn't have a cart
+    # If a user doesn't have a cart, create one
     except NotFoundError:
       return Cart(**{'user': email, 'items': [], 'version': 0})
```

Comment:
tl;dr;not part of the PR, but consider simplifying the complexity of the function `lookup_by_user()` by abstracting the cart creation. In other words, this function currently *needs* to understand *how* to create a new cart (e.g.: the items must be an empty array, and the version must be `0`). Consider abstracting the cart creation to a new function or factory method that initializes all the values. This approach centralizes the logic behind the cart creation, and avoids having to update all the places in the code that create Carts, if the cart creation logic is updated at a later time.

Also, if it makes sense, consider setting default values at the class level during initialization (e.g. items:[], and version:0) instead of keep track of that logic on the calling site.

Reply:
Please see commit [5a94a1d](https://github.com/AndersenLab/CAENDR/pull/403/commits/5a94a1d32af4ba669e0a0893f1754f7018ea0b1c) for changes and let me know if this is what you're looking for.